app/main.py

```diff
@@ -1,6 +1,14 @@
-from typing import Callable
+from collections.abc import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args) -> Callable:
+        if args not in storage:
```

Comment:
In many cases , it is better not to use `not`, because we always thinking if something `is in` but not `is not in`. But a it's a minor thing, that can save you a 3 symbols xD
By the way, it's opinion from my side for this task, sometimes we really need to check `not in` because it looks better with variable names and task requirements.
At that task, if you swap you `if` code and `else` to avoid using not, nothing will change logically!